tests/examples/wibox/template.lua

```diff
@@ -0,0 +1,63 @@
+local file_path, image_path, luacovpath = ...   
+
+-- Set the global shims
+-- luacheck: globals awesome
+awesome = require( "awesome" )
+-- luacheck: globals client
+client  = require( "client"  )
+-- luacheck: globals tag
+tag     = require( "tag"     )
+
+-- Force luacheck to be silent about setting those as unused globals
+assert(awesome and client and tag)
+
+local wibox   = require( "wibox"         )
+local GLib    = require( "lgi"           ).require("GLib")
+local surface = require( "gears.surface" )
+
+-- If luacov is available, use it. Else, do nothing.
+pcall(function()
+    require("luacov.runner")(luacovpath)
+end)
+
+-- Unless there is some kind of shim for that, the Awesome mainloop need to be
+-- emulated.
+local main_loop = GLib.MainLoop()
+
+-- This is the main widget the tests will use as top level
+local container = wibox.layout.fixed.vertical()
+
+-- Let the test request a size and file format
+local w, h, image_type = loadfile(file_path)(container)
+image_type = image_type or "svg"
+
+-- Emulate 10 cycles of mainloop
+local counter = 1
+
+-- Loop as fast as possible
+GLib.timeout_add_seconds(0, 0,function()
```

Comment:
Can you try using `add_idle` (I'm not totally sure about the name). This function will be called whenever there is nothing else to do, which seems better than the 0 second timeout.

Actually... does anything in awesome-lua-land use the GLib main loop? I think you can replace all of this with
```
for counter = 1, 10 do
  awesome:emit_signal("refresh")
end
```
here comes the code you do on the 10th iteration